### PR TITLE
updated

### PR DESCRIPTION
**Pull request by sjpsean**
* changed distance to take double instead of int (I thought user might want to have distance like 1.5km too)

* made private function to handle townData to ensure we fetch only once.

* switched indexes and names for 위도 경도.

### CODE REVIEW

File: example/example.dart [hunk 1]
```diff
@@ -25,26 +25,42 @@ void main() async {
     );
     print(data3);
 
-    // get by lat(위도), lng(경도)
-    List<dynamic> data4 = await nearbyCityService.getNearbyCityByLatLng(
-      lat: 127.069013,
-      lng: 37.316416,
-    );
-    print(data4);
-
```
Comment (sjpsean) [inline example/example.dart:34]: I just pushed this part to the end so I could put examples with the getNearbyCityByLatLng together. What do you think?

File: lib/src/core/core.dart [hunk 2]
```diff
@@ -10,25 +10,34 @@ class Core {
...
 
+  List<dynamic>? _townData;
+
+  Future<List<dynamic>> _getTownData() async {
+    if (_townData == null) {
+      _townData = await _networkRequest.fetchNearbyCity();
+    }
+    return _townData!;
+  }
+
```
Comment (sjpsean) [inline lib/src/core/core.dart:21]: this is where it ensures we fetch data from github only once.

File: lib/src/core/core.dart [hunk 3]
```diff
@@ -10,25 +10,34 @@ class Core {
...
     var townObj = {
       "시도명": nearestTown[0],
       "시군구명": nearestTown[1],
       "읍면동명": nearestTown[2],
-      "위도": nearestTown[3],
-      "경도": nearestTown[4],
+      "경도": nearestTown[3],
+      "위도": nearestTown[4],
       "근처동네": []
     };
```
Comment (sjpsean) [inline lib/src/core/core.dart]: What's your thought on changing the result(townObj) keys to English for future reference (we might be able to use the same function with the townData of other countries. Just an idea.

File: example/example.dart (hunk 1 above)
Comment (jwlim94) [reply example/example.dart:34]: I like that

File: lib/src/core/core.dart (hunk 3 above)
Comment (jwlim94) [reply lib/src/core/core.dart]: Let's do that since it can be extended to US cities

Comment (jwlim94) [reply lib/src/core/core.dart]: 시도명 - county or state?
시군구명 ㅡ city
읍면동명 ㅡ town
경도 ㅡ latitude
위도 ㅡ longitude
거리 ㅡ distance

Comment (sjpsean) [reply lib/src/core/core.dart]: state
city
town
lat
lng
distance
느낌으로 하고 리드미에 한글로 state = 시 도, city = 시군구, town = 읍면동 이런식으로 명시해주면 될듯

Comment (jwlim94) [reply lib/src/core/core.dart]: ['근처동네']도 `nearbyCity` 로 했어

**Pull request merged**